.github/workflows/ci.yml

```diff
@@ -49,4 +49,4 @@ jobs:
         with:
           node-version: 16
           cache: pnpm
-      - run: 'pnpm i && pnpm check && pnpm lint'
+      - run: pnpm i && pnpm check && pnpm lint && pnpm build && { [ "`git status --porcelain=v1`" == "" ] || (echo "Generated types have changed — please run prepublishOnly locally and commit the changes after you have reviewed them"; git diff; exit 1); }
```

Comment:
chaining all checks like this still has the problem that it only shows one error, not all, and you would have to echo for each.  

and if the types are commited, the script to generate them should not be prepublishOnly anymore. best case its a noop, worst case a change slips by